java/src/main/kotlin/io/spine/validation/java/setonce/SetOnceJavaConstraints.kt

```diff
@@ -118,98 +121,108 @@ internal sealed class SetOnceJavaConstraints(
      *
      * Implementation of this method is common for all field types. Inheritors should
      * invoke it within [renderConstraints], passing the necessary parameters.
      *
      * The [currentValue] and [readerStartsWith] are mandatory properties. Pass [readerContains]
      * in cases when [readerStartsWith] is not sufficient. For example, for message fields,
-     * the beginning of the reading block is the same for all fields because it doesn't include
-     * the field name. Differentiation is done way deeper.
+     * the beginning of the reading block is the same for all message fields because it doesn't
+     * include the field name. Differentiation is done way further.
      *
-     * @param currentValue An expression to read the current field value.
-     * @param readerStartsWith An expression representing the beginning of the field reading block.
-     * @param readerContains An arbitrary expression within the field reading block.
+     * @param currentValue The current field value.
+     * @param readerStartsWith The beginning of the field reading block.
+     * @param readerContains An arbitrary code that must be present within the reading block.
      */
     protected fun PsiClass.alterBytesMerge(
-        currentValue: String,
-        readerStartsWith: String,
-        readerContains: String = readerStartsWith,
+        currentValue: Expression<T>,
+        readerStartsWith: JavaElement,
+        readerContains: JavaElement = readerStartsWith,
     ) {
 
         val mergeFromBytes = methodWithSignature(MergeFromBytesSignature).body!!
         val fieldReading = mergeFromBytes.deepSearch(readerStartsWith, readerContains)
-        val fieldProcessing = fieldReading.parent
+        val fieldCaseBlock = fieldReading.parent
 
-        val rememberCurrent = elementFactory.createStatement("var previous = $currentValue;")
-        fieldProcessing.addBefore(rememberCurrent, fieldReading)
+        val previousValue = InitVar("previous", currentValue)
+        fieldCaseBlock.addBefore(previousValue.toPsi(), fieldReading)
 
-        val postcondition = defaultOrSameStatement(
-            currentValue = "previous",
+        val postcondition = throwIfNotDefaultAndNotSame(
+            currentValue = previousValue.read(),
             newValue = currentValue,
         )
-        fieldProcessing.addAfter(postcondition, fieldReading)
+        fieldCaseBlock.addAfter(postcondition, fieldReading)
     }
 
     /**
      * Creates an `if` statement, which checks that the current field value is default
-     * OR if the proposed value is the same as the current.
+     * OR if the proposed [newValue] is the same as the current.
      *
      * Otherwise, it throws the validation exception.
      *
-     * @param currentValue An expression denoting the current field value.
-     * @param newValue An expression denoting the proposed value.
+     * @param currentValue The current field value.
+     * @param newValue The proposed new value.
```

Comment:
Please document `<T>`.

Reply:
In this function and in the one from the comment below, `T` comes from the class definition, where it is already documented. Putting `@param` again yields a warning of undefined parameter.